pkg/poll/poll.go

```diff
@@ -27,37 +28,55 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 )
 
-const (
-	RetryInterval = time.Second * 5
-	Timeout       = time.Second * 30
-)
+//go:generate mockgen -source=poll.go -package=poll -destination=mock_poll_api.go
+
+type PollActions interface {
+	ForResourceUnavailability(obj *unstructured.Unstructured) error
+	ForResource(obj *unstructured.Unstructured) error
+	ForDaemonSet(obj *unstructured.Unstructured) error
+	ForDaemonSetLogs(obj *unstructured.Unstructured, pattern string) error
+}
+
+type pollActions struct {
+	log     logr.Logger
+	waitFor map[string]func(obj *unstructured.Unstructured) error
+}
 
 var (
-	log     = zap.New(zap.UseDevMode(true)).WithName(color.Print("wait", color.Brown))
-	waitFor = map[string]func(obj *unstructured.Unstructured) error{
-		"Pod":                      ForPod,
-		"DaemonSet":                ForDaemonSet,
-		"BuildConfig":              ForBuild,
-		"Secret":                   ForSecret,
-		"CustomResourceDefinition": ForCRD,
-		"Job":                      ForJob,
-		"Deployment":               ForDeployment,
-		"StatefulSet":              ForStatefulSet,
-		"Namespace":                ForResourceAvailability,
-		"Certificates":             ForResourceAvailability,
-	}
+	retryInterval = time.Second * 5
+	timeout       = time.Second * 30
 )
 
+func New() PollActions {
+	actions := pollActions{
+		log: zap.New(zap.UseDevMode(true)).WithName(color.Print("wait", color.Brown)),
+	}
+	waitFor := map[string]func(obj *unstructured.Unstructured) error{
```

Comment:
Does this need to be a new variable? I would move it to the initialization of `actions`.

Reply:
I missed that bit - thanks.